libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -154,6 +136,37 @@ uint32_t ByteBuffer::peekbytes(uint8_t *data, uint32_t len)
     return ret;
 }
 
+int ByteBuffer::reserve(ByteBuffer::IoVec iovec[2], uint32_t len)
+{
+    if (len > space()) {
+        len = space();
+    }
+    if (!len) {
+        return 0;
+    }
+
+    iovec[0].data = &buf[tail];
+    if (tail+len <= size) {
+        iovec[0].len = len;
+    } else {
+        auto n = size - tail;
+        if (n > len) {
+            n = len;
+        }
+
+        iovec[0].len = n;
+
+        tail = (tail + n) % size;
+        n = len - n;
+        if (n > 0) {
```

Comment:
If the code branched to the upper 'else', this should always be true. IOW, it should always have left some bytes that didn't fit the first continuous block, therefore the need for the second iovec buffer.